skafka/src/main/scala/com/evolutiongaming/skafka/JaasConfig.scala

```diff
@@ -0,0 +1,68 @@
+package com.evolutiongaming.skafka
+
+import com.evolutiongaming.config.ConfigHelper.ConfigOps
+import com.evolutiongaming.skafka.ConfigHelpers._
+import com.typesafe.config.{ConfigException, ConfigObject, ConfigValue}
+
+import scala.util.{Failure, Success, Try}
+
+sealed trait JaasConfig {
+  def asString(): String
+}
+
+object JaasConfig {
+
+  private val emptyPath = "\"\""
+
+  final case class Plain(entry: String) extends JaasConfig {
+    override def asString(): String = entry
+  }
+
+  final case class Structured(loginModuleClass: Class[_], controlFlag: String, options: List[Pair]) extends JaasConfig {
+
+    override def asString(): String = s"${loginModuleClass.getName} $controlFlag ${pairAsString()}"
+
+    private def pairAsString() =
+      options
+        .map { case (key, value) => s"$key='$value'" }
+        .mkString("", " ", ";")
+  }
+
+  case object Structured {
+    def make(obj: ConfigObject): Option[Structured] = {
+      val config = obj.toConfig
+      for {
+        loginModuleClass <- config.getOpt[Class[_]]("login-module-class")
+        controlFlag      <- config.getOpt[String]("control-flag")
+        options          <- config.getOpt[List[Pair]]("options")
+      } yield new Structured(loginModuleClass, controlFlag, options)
+    }
+  }
+
+  def apply(config: ConfigValue): JaasConfig = {
```

Comment:
could you please rename this to `fromConfig`

Reply:
Done